### DISCUSSION TITLE
Get device by index from whitelist?

### DISCUSSION THREAD

**Discussion by matthias-bs** [h2zero/NimBLE-Arduino#478] [Q&A] [answered]
Hi,

is there a way to get a device by index by iterating through the whitelist?

I have a whitelist which contains the addresses of all devices I'm interested in, as in https://github.com/h2zero/NimBLE-Arduino/tree/release/1.4/examples/NimBLE_Scan_Whitelist.

It seems I still have to iterate through all devices found to get data from the device I'm looking for, because there is only a function `getWhiteListAddress(i)` (where i is the whitelist-index) but apparently no function like `getWhiteListDevice(i)`.

What I'm doing now is:
```
BLEScanResults foundDevices = _pBLEScan->start(duration, false /* is_continue */);
    
for (unsigned i=0; i<foundDevices.getCount(); i++) {
     if (foundDevices.getDevice(i).getAddress() == BLEAddress(my_sensor)) {
         // Temperature
         int temp_msb = foundDevices.getDevice(i).getServiceData().c_str()[7];
         int temp_lsb = foundDevices.getDevice(i).getServiceData().c_str()[6];
    }
}
```

What I would actually like to do is:
```
for (unsigned i=0; i<NimBLEDevice::getWhiteListCount(); i++) {
         // Temperature
         int temp_msb = NimBLEDevice::getWhiteListDevice(i).getServiceData().c_str()[7];
         int temp_lsb = NimBLEDevice::getWhiteListDevice(i).getServiceData().c_str()[6];
}
```

Or maybe I missed something?

Best regards
Matthias

**Comment by h2zero** [comment, marked as answer]
There is no interface for this since there is no link between the whitelist and the scan results (it's just a filter).

**Comment by matthias-bs** [comment]
Thank you for the answer. Can I get the device from the address then?

**Comment by h2zero** [comment]
There is no interface for this either, sorry.

**Comment by matthias-bs** [comment]
Thanks again, also for implementing this great library!
